src/app/shared/navigation-focus/navigation-focus.ts

```diff
@@ -0,0 +1,40 @@
+import {NgModule, Injectable, OnInit, Directive, Input, ElementRef, Renderer2} from '@angular/core';
+import {Subscription} from 'rxjs/Subscription';
+
+
+export interface NavigationFocusOptions {
+  ignoreFirstLoad?: boolean;
+}
+
+
+@Injectable()
+export class Focuser {
+  focusTimeout: number;
+
+  focus(el: HTMLElement) {
+    window.clearTimeout(this.focusTimeout);
+    this.focusTimeout = window.setTimeout(() => el.focus(), 100);
+  }
+}
```

Comment:
I don't think you need the separate `Focuser` service. We can store the last timeout id in the js module since we know we only ever want there to be one.

```ts
let lastTimeoutId = -1;

@Directive({
  selector: '[focusOnNavigation]',
  host: {'tabindex': '-1'}
})
export class NavigationFocus implements OnInit {
  constructor(private elementRef: ElementRef) {}

  ngOnInit() {
    clearTimeout(lastTimeoutId);
    lastTimeoutId = setTimeout(() => elementRef.nativeElement.focus(), 100);
  }
}
```
(also baked in a few style suggestions there)